proxy/reverse.go

```diff
@@ -134,6 +135,11 @@ func (p *reverseProxy) ServeHTTP(rw http.ResponseWriter, clientreq *http.Request
 		}
 		redirectRequest(proxyreq, ep.URL)
 
+		if repeated := p.checkProxyLoop(proxyreq, ep.URL.String()); repeated {
```

Comment:
If local proxyID appears in the header of incoming http request, this indicates that something wrong has happened, and the associated endpoint should be banned. So there is no need to wait until proxying phase to detect it IMO.